### PR TITLE
Remove Yuni::Bind from src, excluding src/ui

### PR DESCRIPTION
**Pull request by flomnes**
Remove `Yuni::Bind`, use `std::function`, `std::bind` instead.

Don't do it in src/ui (deprecated component).

### CODE REVIEW

File: src/ui/simulator/toolbox/components/map/settings.h [hunk 1]
```diff
@@ -25,6 +25,7 @@
 #include <wx/bitmap.h>
 #include <wx/image.h>
 #include <list>
+#include <cstdint>
```
Comment (flomnes) [inline src/ui/simulator/toolbox/components/map/settings.h:28]: See `uint16_t` below

File: src/ui/simulator/windows/cleaner.cpp [hunk 2]
```diff
@@ -57,7 +57,7 @@ class CleaningThread final : public Yuni::Thread::IThread
         {
             std::lock_guard locker(pParent->pMutex);
             if (pParent->pInfos)
-                pParent->pInfos->onProgress.unbind();
+                pParent->pInfos->onProgress = nullptr;
```
Comment (flomnes) [inline src/ui/simulator/windows/cleaner.cpp]: Need to check what happens if we call an unbinded Yuni::Bind function. Calling a nullptr std::function is UB (?)

Comment (flomnes) [reply src/ui/simulator/windows/cleaner.cpp]: It seems that calling a non-bound `Yuni::Bind` is a no-op
```cpp
// Bind: unbind
template<class R>
inline void Bind<R(), void>::unbind()
{
    pHolder = new Private::BindImpl::None<R()>(); // unbind
}

template<class R>
inline R Bind<R(), void>::operator()() const
{
    return pHolder->invoke();
}
```

Comment (flomnes) [reply src/ui/simulator/windows/cleaner.cpp]: Done, will use `[]{}`, see https://stackoverflow.com/a/2982599

**Review by JasonMarechal25 — APPROVED** [review]

**Pull request merged**